### DISCUSSION TITLE
Defining ENV variables with JS at build time

### DISCUSSION THREAD

**Discussion by iDm1** [parcel-bundler/parcel#9035] [General]
With Webpack I can use DefinePlugin to define ENV variable at build time with JS:
```js
new webpack.DefinePlugin({
     BUILD_TIME: new Date().toString()
});
```
According to Parcel documentation it's impossible to achieve without a transformer plugin. But who knows what plugin
 does this? There is "parcel-transformer-inject-env", but it does nothing without Plasmo Framework.

How such a simple task is usually done with Parcel?

**Comment by joshamaju** [comment]
If you're using the parcel API, you can pass down environment variables

**Comment by davidjb** [comment]
I'm also keen to know if this is possible dynamically at build time. As has been mentioned above, it's possible to pass `env` when creating an instance of `Parcel` (https://parceljs.org/features/parcel-api/#environment-variables), but these are set and (seemingly) unable to be changed.

In order to create a build with different environment variables this seems to require creating a new `Parcel` instance. In the case of a watcher running, changing an environment variable means tearing down the watcher and restarting it, a costly operation.

Is it possible to pass ENV variables (or just arbitrary data) into a build at build time, or is it otherwise possible to change a `Parcel`'s ENV variables on the fly?
